apis/extension/descheduling.go

```diff
@@ -34,6 +36,20 @@ const (
 	AnnotationEvictionCost = SchedulingDomainPrefix + "/eviction-cost"
 )
 
+const (
+	// AnnotationEvictionMark indicates custom eviction. It can be used to set to an "true".
+	AnnotationEvictionMark = SchedulingDomainPrefix + "/custom-eviction"
```

Comment:
Maybe we can directly set the annotation key to `koordinator.sh/eviction`, and the value of this annotation is the specific field of the eviction request.



 For example:
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
      koordinator.sh/eviction: |-
          {
               "timestamp": "the timestamp of eviction requests, and the format is metav1.Time",
               "deleteOptions":  "metav1.DeleteOptions",
               "initiator": "We need this field to help quickly locate the initiator to judge whether the eviction is reasonable", 
               "reason": "This information is also useful",
               "controller": "There are some complex scenarios where different workloads are evicted or deleted by different controllers. So we need this field. It is not set by default, and when it is not set, a default controller can be responsible for eviction."
          }
```

WDYT @leason00 cc @hormes @FillZpp

Reply:
Except for `controller`, everything else seems to be fine. From the above description, the controller needs to be configured by the user, or declared on the Pod, to determine whether theirs custom controller will take over and evict the annotation Pod

Reply:
Yes. Your description is more accurate. The controller should indeed be specified by the user on the Pod, not by our descheduler.